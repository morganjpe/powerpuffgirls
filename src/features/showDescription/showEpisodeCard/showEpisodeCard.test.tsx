import { render, screen } from '@testing-library/react';
import ShowEpisodeCard from './showEpisodeCard';

describe('<ShowEpisodeCard /> component', () => {
  it('should display a card with image, title and episode number', () => {
    render(
      <ShowEpisodeCard
        title="episode name"
        image={
          'https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg'
        }
        number={10}
      />
    );

    const image = screen.getByAltText(/episode name/i);
    expect(image).toHaveAttribute(
      'src',
      'https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg'
    );

    screen.getByText(/episode name/i);
    screen.getByText(/10/i);
  });
});
