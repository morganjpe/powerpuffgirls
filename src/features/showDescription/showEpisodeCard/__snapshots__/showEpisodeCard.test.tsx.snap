// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowEpisodeCard /> component should display a card with image, title and episode number 1`] = `
"<div class=\\"sc-gtsrHT bVWcXv\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg\\" alt=\\"episode name\\">
  <div>
    <h4>episode name <span>10</span></h4>
  </div>
</div>"
`;

exports[`<ShowEpisodeCard /> component should render a placeholder if no image 1`] = `
"<div class=\\"sc-gtsrHT bVWcXv\\">
  <div width=\\"0\\" class=\\"sc-bdnxRM hKWRKm\\">no image thumbnail</div>
  <div>
    <h4>episode name <span>10</span></h4>
  </div>
</div>"
`;
